presto-spi/src/main/java/com/facebook/presto/spi/type/UnscaledDecimal128Arithmetic.java

```diff
@@ -730,6 +730,117 @@ static void shiftRight(Slice decimal, int rightShifts, boolean roundUp, Slice re
         pack(result, lo, hi, negative);
     }
 
+    public static Slice divideRoundUp(long dividend, int dividendScaleFactor, long divisor)
```

Comment:
I would put those wrapper methods in `DecimalOperators.java` as it is done with other operators. That would leave only the important function ` private static Slice divideRoundUp(long dividendLow, long dividendHi, int dividendScaleFactor, long divisorLow, long divisorHi)` here.